src/Common/src/System/Net/WebSockets/WebSocketValidate.cs

```diff
@@ -123,9 +124,19 @@ internal static void ThrowPlatformNotSupportedException()
 
         internal static void ValidateArraySegment(ArraySegment<byte> arraySegment, string parameterName)
         {
+            Debug.Assert(!string.IsNullOrEmpty(parameterName), "'parameterName' MUST NOT be NULL or string.Empty");
+
             if (arraySegment.Array == null)
             {
-                throw new ArgumentNullException(parameterName + ".Array");
+                throw new ArgumentNullException(parameterName + "." + nameof(arraySegment.Array));
+            }
+            if (arraySegment.Offset < 0 || arraySegment.Offset > arraySegment.Array.Length)
```

Comment:
That's why these checks were previously removed.  Though it is possible.  See the trick employed here: https://github.com/dotnet/corefx/blob/773b2573afc047717d1366f21a9c3fb0d9e05f51/src/System.Net.Sockets/tests/FunctionalTests/SendReceive.cs#L52